libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts

```diff
@@ -191,22 +192,32 @@ export class ResourceExplorer {
             throw new Error(`Resource ${typeof resourceOrId === 'string' ? resourceOrId : resourceOrId.id} not found.`);
         }
 
-        const json = resource.readText();
-        const result = JSON.parse(json, (_key: string, value: unknown): unknown => {
-            if (typeof value !== 'object') {
-                return value;
-            }
-            if (Array.isArray(value)) {
-                return value;
-            }
-            const kind = value['$kind'];
-            if (!kind) {
-                return value;
-            }
-            return this.load(value as { $kind: string } & Record<string, unknown>);
-        }) as T;
+        if (this._cache.has(resource.id)) {
+            return this._cache.get(resource.id) as T;
```

Comment:
I think this is where you might have to throw if cycle detection is explicitly disabled.